angular2/src/angular2TextMask.ts

```diff
@@ -33,6 +33,7 @@ export default class MaskedInputDirective {
   }
 
   writeValue(value: any) {
+    this.textMaskInputElement.update(value)
```

Comment:
Ok, I should have waited until morning before I typed my response, couldn't remember exactly what was happening. I had to put that in the `writeValue` function because it's the only place I could find to access the model that was coming into the directive at load time and the `textMask` was overwriting with the placeholder. If we can find a way to access it `onInit` I think that might be better but not sure how. The other solution might be to add logic to the `writeValue` method that checks against a variable and then would only call the `.update()` method the first time. What do you think, worth it if you're already catching duplicate calls?

Also, I don't think line 52 is needed anymore. It seemed to work fine without it but I wasn't sure if there was an edge case you had added that for so I left it. But now that I think about it, I don't think you'd ever want that to run if the value didn't get updated properly from the initial writeValue anyway.